extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -132,4 +172,34 @@ export class ScrollManager {
       this.onScroll_.bind(this)
     );
   }
+
+  /** @return {!Element} */
+  getRootElement_() {
+    const root = this.ampdoc_.getRootNode();
+    return dev().assertElement(
+      // In the case of a shadow doc, its host will be used as
+      // a refrence point, otherwise use the same implementation
+      // as in `analytics-root.js`
+      root.host || root.documentElement || root.body || root
+    );
+  }
+
+  /**
+   * Gets the layout rectangle of the root element
+   * @param {cached=} initial
+   * @return {!Promise<!../../../src/layout-rect.LayoutRectDef>}
+   */
+  measureRootElement_(initial = false) {
```

Comment:
Let's split into `getOrMeasureInitialRootRect_` and `measureRootRect_`. No boolean args.

```js
getOrMeasureInitialRootRect_() {
  if (this.initialRootLayoutRect_) {
     return Promise.resolve(this.initialRootLayoutRect_);
  }
  return this.measureRootRect_();
}
measureRootRect_() {
  return this.mutator_.measureElement(() => {
    const layoutRect = this.viewport_.getLayoutRect(this.root_);
    if (!this.initialRootLayoutRect_) {
      this.initialRootLayoutRect_ = layoutRect;
    }
    return layoutRect;
  });
}
```